tests/test_message.py

```diff
@@ -1,67 +1,105 @@
 
-    def test_parse_type_own_message(self):
-        message = Message.parse(TestMessage.raw_sync_message)
+    async def test_parse_type_own_message(self):
+        message = await Message.parse(self.signal_api, TestMessage.raw_sync_message)
         self.assertEqual(message.type, MessageType.SYNC_MESSAGE)
 
-    def test_parse_text_own_message(self):
-        message = Message.parse(TestMessage.raw_sync_message)
+    async def test_parse_text_own_message(self):
+        message = await Message.parse(self.signal_api, TestMessage.raw_sync_message)
         self.assertEqual(message.text, TestMessage.expected_text)
 
-    def test_parse_group_own_message(self):
-        message = Message.parse(TestMessage.raw_sync_message)
+    async def test_parse_group_own_message(self):
+        message = await Message.parse(self.signal_api, TestMessage.raw_sync_message)
         self.assertEqual(message.group, TestMessage.expected_group)
 
     # Foreign Messages
-    def test_parse_source_foreign_message(self):
-        message = Message.parse(TestMessage.raw_data_message)
+    async def test_parse_source_foreign_message(self):
+        message = await Message.parse(self.signal_api, TestMessage.raw_data_message)
         self.assertEqual(message.timestamp, TestMessage.expected_timestamp)
 
-    def test_parse_timestamp_foreign_message(self):
-        message = Message.parse(TestMessage.raw_data_message)
+    async def test_parse_timestamp_foreign_message(self):
+        message = await Message.parse(self.signal_api, TestMessage.raw_data_message)
         self.assertEqual(message.source, TestMessage.expected_source)
 
-    def test_parse_type_foreign_message(self):
-        message = Message.parse(TestMessage.raw_data_message)
+    async def test_parse_type_foreign_message(self):
+        message = await Message.parse(self.signal_api, TestMessage.raw_data_message)
         self.assertEqual(message.type, MessageType.DATA_MESSAGE)
 
-    def test_parse_text_foreign_message(self):
-        message = Message.parse(TestMessage.raw_data_message)
+    async def test_parse_text_foreign_message(self):
+        message = await Message.parse(self.signal_api, TestMessage.raw_data_message)
         self.assertEqual(message.text, TestMessage.expected_text)
 
-    def test_parse_group_foreign_message(self):
-        message = Message.parse(TestMessage.raw_data_message)
+    async def test_parse_group_foreign_message(self):
+        message = await Message.parse(self.signal_api, TestMessage.raw_data_message)
         self.assertEqual(message.group, TestMessage.expected_group)
 
-    def test_read_reaction(self):
-        message = Message.parse(TestMessage.raw_reaction_message)
+    async def test_read_reaction(self):
+        message = await Message.parse(self.signal_api, TestMessage.raw_reaction_message)
         self.assertEqual(message.reaction, "👍")
 
+    @patch("aiohttp.ClientSession.get", new_callable=AsyncMock)
+    async def test_attachments(self, mock_get):
```

Comment:
This test is not enough, also add a message with an attachment to https://github.com/filipre/signalbot/blob/master/tests/test_api.py#L36C15-L36C27 and https://github.com/filipre/signalbot/blob/master/tests/test_bot.py#L25

Reply:
They payload for the sent messages is not checked. Adding tests for it is out of scope for this PR.